### PR TITLE
sanity testing normal driver

### PR DESCRIPTION
**Pull request by pohly**
This enhances the driver such that it passes testing with csi-sanity from csi-test v2.0.0 and improves testing.

### CODE REVIEW

File: deploy/kubernetes-1.13/pmem-csi-lvm.yaml [hunk 1]
```diff
@@ -98,7 +98,19 @@ spec:
   selector:
     app: pmem-csi-controller
   ports:
-  - port: 10000
+  - port: 10000 # port inside the pod where the PMEM-CSI registry listens
+---
+# TODO: only deploy this for testing
```
Comment (pohly) [inline deploy/kubernetes-1.13/pmem-csi-lvm.yaml]: This is the part where I'd like to get feedback. In order to test the normal driver deployment, we have to expose it's csi.sock to the outside world. This is something that must not be done in production.

We need to investigate how we can minimize the number of different deployment files (lvm/direct, with/without socat, perhaps support different log levels).

In the meantime my proposal is to just add these socat parts here and also add them to pmem-csi-direct.yaml (currently missing there).

Comment (pohly) [reply deploy/kubernetes-1.13/pmem-csi-lvm.yaml]: I think I have found a clean solution after rebasing this branch onto PR #234: now the `-testing` variants are deployed in our test cluster and provides the necessary port exposure, whereas the normal deployments are completely unchanged.

Therefore I now consider this PR ready for merging.

**Comment by pohly** [comment]
@okartau this PR is a superset of PR #234.

File: README.md [hunk 2]
```diff
@@ -502,6 +489,30 @@ The `deploy` directory contains one directory or symlink for each
 tested Kubernetes release. The most recent one might also work on
 future, currently untested releases.
 
+- **Wait until all pods reach 'Running' status**
+
+```sh
+    $ kubectl get pods
+    NAME                    READY   STATUS    RESTARTS   AGE
+    pmem-csi-8kmxf          2/2     Running   0          3m15s
+    pmem-csi-bvx7m          2/2     Running   2          3m15s
```
Comment (avalluri) [inline README.md]: Why this driver restarts visible again :( ?

Comment (okartau) [reply README.md]: not sure. When I made README addition, there was this text. I think this was Clear-based cluster. I haven't re-checked this properly. But as positive example, ubuntu-govm based cluster I develop right now under #230, shows clean state:
```
NAME                    READY   STATUS    RESTARTS   AGE
pmem-csi-controller-0   3/3     Running   0          8m46s
pmem-csi-node-5vx7t     2/2     Running   0          7m24s
pmem-csi-node-bqlbl     2/2     Running   0          7m51s
pmem-csi-node-ctjrq     2/2     Running   0          7m6s
```

I think this topic needs checking

Comment (pohly) [reply README.md]: This README.md change isn't supposed to be in this PR at all.

Comment (pohly) [reply README.md]: Let's get PR #234 merged and then I'll rebase.

Comment (pohly) [reply README.md]: This spurious change was the result of incomplete rebasing. It is gone now.

**Comment by pohly** [comment]
Rebased. When merging this, please use "Merge pull request", not "Rebase merge", because I have other branches based on this and rebasing the base leads to problems.

**Pull request merged**